### PR TITLE
Singlezone setpoint managers - correct no flow setpoint

### PR DESCRIPTION
**Pull request by mjwitte**
Pull request overview
---------------------
Partially addresses #7145.
SetpointManager:SingleZone:Heating and SetpointManager:SingleZone:Cooling were setting the wrong setpoint if the system had zero flow at the time the setpoint manager was executed. At the time that the setpoint managers are executed, the mass flow is from the end of the prior timestep.  This primarily caused problems for timesteps in which a night-cycle availability manager (or some other trigger) caused the system to run after a period of being off.

SetpointManager:SingleZone:Heating now sets the minimum setpoint for no flow, and SetpointManager:SingleZone:Cooling now sets the maximum setpoint.

Diffs
--------------
SmOffPSZ-MultiModeDX - some significant changes at the timestep level, but an annual simulation has <0.1% change in electric and gas use and identical hours not met.
AirflowNetwork_MultiAirLoops - similar, tiny change in annual results.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

**Review by rraustad — APPROVED** [review]
Same issue I have seen in the past with identical code change.

**Review by EnergyArchmage — APPROVED** [review]
I have reviewed these changes in January.  Good catch to contribute back to DOE version.

**Comment by rraustad** [comment]
The result of changing the no-flow set point initialization is as expected. Heating coil operating is limited to winter months and coils do not turn on at the same time. Note x-axis shows down-shifting occurs less often with the detailed reporting frequency. Defect file didn't have zone temp and didn't think it necessary to run these files again. Zone temp would have previously been negatively impacted.

image

**Comment by rraustad** [comment]
@mjwitte are you set up to look at the failed unit test? I'm on this branch now.

**Comment by mjwitte** [comment]
@rraustad The failed unit test is another red herring on CI - it randomly fails on Mac only. Edwin has a story to look at it. I'm working on a new unit test for this, will push soon.

File: src/EnergyPlus/SetPointManager.cc [hunk 1]
```diff
@@ -5547,28 +5547,10 @@ namespace SetPointManager {
         // From the heating load of the control zone, calculate the supply air setpoint
         // needed to meet that zone load (based on CalcSingZoneRhSetPoint)
 
-        // METHODOLOGY EMPLOYED:
-        // na
-
-        // REFERENCES:
-        // na
-
-        // Using/Aliasing
```
Comment (Myoldmopar) [inline src/EnergyPlus/SetPointManager.cc:5556]: :+1: cleaning up empty comments :+1:

File: src/EnergyPlus/SetPointManager.cc [hunk 2]
```diff
@@ -5583,18 +5565,14 @@ namespace SetPointManager {
         ZoneMassFlow = Node(ZoneInletNode).MassFlowRate;
         ZoneLoadtoHeatSP = ZoneSysEnergyDemand(ZoneNum).OutputRequiredToHeatingSP;
         ZoneTemp = Node(ZoneNode).Temp;
-        // CR7654 IF (ZoneLoadtoHeatSP.GT.0.0) THEN
         if (ZoneMassFlow <= SmallMassFlow) {
-            this->SetPt = this->MaxSetTemp;
+            this->SetPt = this->MinSetTemp;
```
Comment (Myoldmopar) [inline src/EnergyPlus/SetPointManager.cc:5569]: So for heating we'll set the minimum now.  Ok.

File: src/EnergyPlus/SetPointManager.cc [hunk 3]
```diff
@@ -5647,18 +5607,14 @@ namespace SetPointManager {
         ZoneMassFlow = Node(ZoneInletNode).MassFlowRate;
         ZoneLoadtoCoolSP = ZoneSysEnergyDemand(ZoneNum).OutputRequiredToCoolingSP;
         ZoneTemp = Node(ZoneNode).Temp;
-        // CR7654 IF (ZoneLoadtoCoolSP.LT.0.0) THEN
         if (ZoneMassFlow <= SmallMassFlow) {
-            this->SetPt = this->MinSetTemp;
+            this->SetPt = this->MaxSetTemp;
```
Comment (Myoldmopar) [inline src/EnergyPlus/SetPointManager.cc:5611]: And for cooling if no/low flow we set it to maximum temp.  Ok.

File: tst/EnergyPlus/unit/SetPointManager.unit.cc [hunk 4]
```diff
@@ -1419,3 +1419,109 @@ TEST_F(EnergyPlusFixture, SingZoneCoolHeatSetPtMgrZoneInletNodeTest)
...
+    EXPECT_NEAR(heatSPNode.TempSetPoint, 19.01, 0.01);
+
+    // Case 5 - Large cooling load
+    inletNode.MassFlowRate = 0.1;
+    zoneNode.Temp = 20.0;
+    DataZoneEnergyDemands::ZoneSysEnergyDemand(1).OutputRequiredToCoolingSP = -5000.0;
+    DataZoneEnergyDemands::ZoneSysEnergyDemand(1).OutputRequiredToHeatingSP = -20000.0;
+    SetPointManager::ManageSetPoints();
+    EXPECT_NEAR(coolSPNode.TempSetPoint, 14.0, 0.01);
+    EXPECT_NEAR(heatSPNode.TempSetPoint, -99.0, 0.01);
```
Comment (Myoldmopar) [inline tst/EnergyPlus/unit/SetPointManager.unit.cc:1526]: This is a great unit test! :+1:

**Review by Myoldmopar — APPROVED** [review]
Code changes look to be in accordance with the existing fix.  Unit test is great.

**Comment by Myoldmopar** [comment]
Everything here looks great, and there are 3 approvals.  Merging.

**Pull request merged**